core/src/apps/bitcoin/sign_tx/layout.py

```diff
@@ -183,13 +183,23 @@ async def confirm_total(
     ctx: wire.Context,
     spending: int,
     fee: int,
+    fee_rate: float,
     coin: CoinInfo,
     amount_unit: AmountUnit,
 ) -> None:
+    fee_rate_str = ""
+
+    if fee_rate >= 0:
+        # Modify `fee_rate` for `format_amount` function (multiply by 10, because
+        # we only want to display 1 decimal digit and then get whole number)
+        fee_rate_multiplied = int(fee_rate * 10)
+        fee_rate_str = f"\n({format_amount(fee_rate_multiplied, 1)} sat/vB)"
```

Comment:
Yes, I could, but I wanted to use the "format" function because it is already implemented and used by other functions (so we have common code with same behaviour). Also "format_*" functions are used throughout the code in this source file, so I wanted to follow this practice as well.

On the other hand this code would be much shorter and simpler.

Reply:
The purpose of `format_amount` is to work with fixed-point decimal values, where you get an amount in sats (or equivalent) and the position of the fixed point. There is no built-in way to work with this kind of value in micropython.
but fee rate is naturally floating-point (or in other words arbitrary precision) so it doesn't seem appropriate and we can use the built-in floating point formatting facility.

I am not sure if the same argument applies to legacy code, because I'm not sure if we have readily available floating point formatting there.

Reply:
separate point: the `\n` and `()` should not be part of the passed in string. instead, the layout function should take care of inserting the line break and formatting the whole layout as it sees fit

Reply:
5c63c50f54a1a59289bda6176cc97d293950152e